modules/contextualization/cdf_file_annotation/functions/fn_file_annotation_finalize/services/ApplyService.py

```diff
@@ -0,0 +1,341 @@
+import abc
+import json
+from datetime import datetime, timezone
+from hashlib import sha256
+from typing import Any, cast
+
+from cognite.client import CogniteClient
+from cognite.client.data_classes import RowWrite
+from cognite.client.data_classes.data_modeling import (
+    DirectRelationReference,
+    ViewId,
+    EdgeApply,
+    NodeOrEdgeData,
+    Node,
+    NodeId,
+    NodeApply,
+    NodeApplyResultList,
+    EdgeId,
+    InstancesApplyResult,
+)
+
+from cognite.client.data_classes.filters import (
+    In,
+    Or,
+)
+
+
+from services.ConfigService import Config, ViewPropertyConfig
+from utils.DataStructures import DiagramAnnotationStatus
+from services.LoggerService import CogniteFunctionLogger
+
+
+class IApplyService(abc.ABC):

```

Comment:
Only the function below uses this interface. integrate it into the class. Its adding complexity for no value